plugins/restful/RestfulFilesUpload.php

```diff
@@ -26,6 +26,8 @@ public static function controllersInfo() {
    * - "validators": By default no validation is done on the file extensions or
    *   file size.
    * - "scheme": By default the default scheme (e.g. public, private) is used.
+   * - "make_permanent": When true the file's status will be set to permanent
```

Comment:
>  it's also the file reference which will prevent it from being deleted.

IMO, the file reference is an implementation detail to the fact you want the file to be permanent or temporary. By using the named constant `FILE_STATUS_PERMANENT` I think it makes it more readable and more inline with core's lingo.